# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/ScrollStyles.cpp
# Commit: 9b96152c5e6e
# Full Hash: 9b96152c5e6ecf97af63865f357e8106e0763ea4
# Author: Hiroyuki Ikezoe <hikezoe@mozilla.com>
# Date: 2019-06-27 09:34:48
# Regressor Bug: 1560237
# File Overlap Count: 1
# Description:
#   Bug 1560237 - Don't propagate scroll-behavior from <body>. r=botond
#   
#   From the CSSOM View spec[1];
#   
#    The scroll-behavior property of the HTML body element is not propagated to
# ==============================================================================

diff -r a269192c5d30 -r 9b96152c5e6e layout/base/ScrollStyles.cpp
--- a/layout/base/ScrollStyles.cpp	Wed Jun 26 20:59:14 2019 +0000
+++ b/layout/base/ScrollStyles.cpp	Wed Jun 26 20:57:05 2019 +0000
@@ -51,7 +51,6 @@
                            StyleOverflow aV, const nsStyleDisplay* aDisplay)
     : mHorizontal(aH),
       mVertical(aV),
-      mScrollBehavior(aDisplay->mScrollBehavior),
       mOverscrollBehaviorX(aDisplay->mOverscrollBehaviorX),
       mOverscrollBehaviorY(aDisplay->mOverscrollBehaviorY) {
   InitializeScrollSnapStrictness(aWritingMode, aDisplay);
@@ -61,7 +60,6 @@
                            const nsStyleDisplay* aDisplay)
     : mHorizontal(aDisplay->mOverflowX),
       mVertical(aDisplay->mOverflowY),
-      mScrollBehavior(aDisplay->mScrollBehavior),
       mOverscrollBehaviorX(aDisplay->mOverscrollBehaviorX),
       mOverscrollBehaviorY(aDisplay->mOverscrollBehaviorY) {
   InitializeScrollSnapStrictness(aWritingMode, aDisplay);